tests/phpunit/tests/db.php

```diff
@@ -2472,4 +2472,91 @@ public function data_parse_db_host() {
 			),
 		);
 	}
+
+	/**
+	 * @ticket 55414
+	 */
+	public function test_suppress_query_error() {
+		global $wpdb;
+
+		$old_show_errors     = $wpdb->show_errors;
+		$old_suppress_errors = $wpdb->suppress_errors;
+		$old_last_query      = $wpdb->last_query;
+		$old_error_log_path  = ini_get( 'error_log' );
+
+		$suppress_query_error_filter = new MockAction();
+		$log_query_error_action      = new MockAction();
+
+		try {
+			// Redirect error logging so this test doesn't print messages in the test output.
+			$temp_error_log = tmpfile();
+			$this->assertNotEmpty( $temp_error_log, 'Unable to create temp file for error logging' );
+			$temp_error_log_meta = stream_get_meta_data( $temp_error_log );
+			ini_set( 'error_log', $temp_error_log_meta['uri'] );
+
+			// Disable `show_errors` to avoid being labeled a risky test.
+			$wpdb->show_errors = false;
+			$wpdb->last_query  = 'expected-query';
+
+			add_filter( 'suppress_query_error', array( $suppress_query_error_filter, 'filter' ), 10, 3 );
+			add_filter( 'log_query_error', array( $log_query_error_action, 'action' ), 10 );
+
+			// Test that the current suppress_errors setting is passed to filter.
+			$wpdb->suppress_errors = false;
+			$wpdb->print_error( 'expected-error' );
+			$this->assertSame( 1, $suppress_query_error_filter->get_call_count() );
+			$this->assertSame(
+				$suppress_query_error_filter->events[0]['args'],
+				array( false, 'expected-error', 'expected-query' )
+			);
+			$this->assertSame( 1, $log_query_error_action->get_call_count() );
+
+			$suppress_query_error_filter->reset();
+			$log_query_error_action->reset();
+			$wpdb->suppress_errors = true;
+			$wpdb->print_error( 'expected-error' );
+			$this->assertSame( 1, $suppress_query_error_filter->get_call_count() );
+			$this->assertSame(
+				$suppress_query_error_filter->events[0]['args'],
+				array( true, 'expected-error', 'expected-query' )
+			);
+			$this->assertSame( 0, $log_query_error_action->get_call_count() );
+
+			// Test altering error suppression with the filter.
+			$suppress_query_error_filter->reset();
+			$log_query_error_action->reset();
+			$wpdb->suppress_errors = true;
+			add_filter( 'suppress_query_error', '__return_false', 9 );
+			$wpdb->print_error( 'expected-error' );
+			remove_filter( 'suppress_query_error', '__return_false', 9 );
+			$this->assertSame( 1, $suppress_query_error_filter->get_call_count() );
+			$this->assertSame(
+				$suppress_query_error_filter->events[0]['args'],
+				array( false, 'expected-error', 'expected-query' )
+			);
+			$this->assertSame( 1, $log_query_error_action->get_call_count() );
+
+			$suppress_query_error_filter->reset();
+			$log_query_error_action->reset();
+			$wpdb->suppress_errors = false;
+			add_filter( 'suppress_query_error', '__return_true', 9 );
+			$wpdb->print_error( 'expected-error' );
+			remove_filter( 'suppress_query_error', '__return_true', 9 );
```

Comment:
The `tear_down()` method in the test suite clears up hooks between tests, so once these tests are separated and use a data provider, these `remove_filter()` calls will likely be unnecessary.

Reply:
That's great to know. The tests are now separated and using data providers so the removal calls have been removed. :)